ec/helper/tf.go

```diff
@@ -172,6 +173,37 @@ func ConvertToStringMapPointer(v interface{}, excludeEmpty bool) *map[string]str
 	return nil
 }
 
+// StringIsNotEmptyJSON is a custom validator function that can be used as a
+// schema attribute validator function.
+func StringIsNotEmptyJSON(
+	i interface{},
+	k string,
+) (warnings []string, errors []error) {
+	v, ok := i.(string)
+	if !ok {
+		errors = append(
+			errors,
+			fmt.Errorf("expected type of %s to be string", k))
```

Comment:
I might be reading this wrong, but the type assertion is on "i" but the error included is "k". Did you mean to return something else as the error message?

Reply:
@wdawg33 I think the variable is unclear (it's a direct copy from other Hashicorp validation code) 

k stands for key - the property being validated. So for example it will be "policy" as in the policy is expected to be a string. 

I'm open to a more helpful error message, I was probably going to do a second pass on this code anyway. Perhaps it could include the actual type with %T?

Reply:
Understood, thanks for the clarification. If you are going to take a second pass this is definitely good as is. Thanks!